webapp/src/app.tsx

```diff
@@ -1881,6 +1882,24 @@ export class ProjectView
         return Promise.resolve();
     }
 
+    async resetTutorialTemplateCode(keepAssets: boolean): Promise<void> {
+        const mainPkg = pkg.mainEditorPkg();
+        const header = mainPkg.header;
+        if (!header?.tutorial?.templateCode) return;
+
+        if (keepAssets) {
+            // Convert all temporary assets to named assets before we load in the template
+            let currentText = await workspace.getTextAsync(header.id);
+            const imageJres = appendTemporaryAssets(currentText[pxt.MAIN_BLOCKS], currentText[pxt.IMAGES_JRES]);
+            pkg.mainEditorPkg().setFile(pxt.IMAGES_JRES, imageJres);
+            await mainPkg.saveFilesAsync();
+        }
+
+        header.tutorial.templateLoaded = false;
+        delete header.tutorial.mergeHeaderId;
```

Comment:
what happens if they restart the tutorial a second time? is that possible?

Reply:
yeah, when they restart it re-imports the code so the merge id is reset and there shouldn't be any issue (the mergeProjects call happens after the template code check, so previously it only ran once per project anyway). they can also reset the template code more than once and that shouldn't have any issues